torch/_higher_order_ops/schema.py

```diff
@@ -201,13 +222,51 @@ def from_hop_argument_info(
                 for i, val in enumerate(out_argument_info.example_value)
             ]
 
-        return torch._C.FunctionSchema(
+        return HopSchema(
             op_name,
             "",
             args,
             rets,
             False,
             False,
+            schema_tree_spec,
+        )
+
+
+class HopSchema(torch._C.FunctionSchema):
+    def __init__(
+        self,
+        name: str,
+        overload_name: str,
+        arguments: list[torch._C.Argument],
+        returns: list[torch._C.Argument],
+        is_vararg: bool,
+        is_varret: bool,
+        schema_tree_spec: pytree.TreeSpec,
+    ):
+        self.tree_spec = schema_tree_spec
```

Comment:
Can this ever be None or does it always need to be populated?
For schemas that don't need the flatten/unflatten, is this a LeafSpec?